chutney/ui/src/app/modules/campaign/components/execution/history/campaign-executions-history.component.ts

```diff
@@ -128,7 +129,7 @@ export class CampaignExecutionsHistoryComponent implements OnInit, OnDestroy {
     }
 
     private refreshCampaign() {
-        if (this.campaign.scenarios.length > 0) {
+        if (this.isStillOnThePage() && this.campaign.scenarios.length > 0) {
```

Comment:
I prefer to handle it differently to avoid background events after leaving compaign page. Possible solutions:

- review ` eventManagerService.subscribe`: 
    
```
subscribe(eventName, callback) {
        let isUnsubscribed = false;

        const subscription = this.observable
            .pipe(filter(event => event.name === eventName))
            .subscribe(event => {
                if (!isUnsubscribed) callback(event);
            });

        return {
            unsubscribe: () => {
                isUnsubscribed = true;
                subscription.unsubscribe();
            }
        };
    }
```
and then use it like:
```
const subscription = subscribe('eventName', event => {
    console.log(event);
});

// Later
subscription.unsubscribe();
```

- The best fix would be to review `eventManagerService.subscribe` implementation to return an observable(or `signal`) instead of subscription and remove callback param. Then when calling it do something like:

```
 this.onExecuteSubscription = this.eventManagerService.subscribe('execute').pipe(
            switchMap(() =>  this.refreshCampaign())
        );
``` 
and review `refreshCampaign` to return `observable` instead of `void`.